core/src/test/scala/com/velocidi/apso/elasticsearch/ElasticsearchBulkInserterSpec.scala

```diff
@@ -0,0 +1,102 @@
+package com.velocidi.apso.elasticsearch
+
+import scala.concurrent.duration._
+
+import akka.actor._
+import akka.testkit._
+import com.sksamuel.elastic4s.ElasticDsl._
+import com.sksamuel.elastic4s.requests.searches._
+import io.circe.Json
+import io.circe.syntax._
+import net.ruippeixotog.akka.testkit.specs2.mutable.AkkaSpecification
+import org.specs2.concurrent.ExecutionEnv
+
+import com.velocidi.apso.elasticsearch.ElasticsearchBulkInserter.{ ElasticsearchDown, ElasticsearchUp, Insert }
+import com.velocidi.apso.elasticsearch.config.Elasticsearch
+
+class ElasticsearchBulkInserterSpec(implicit ee: ExecutionEnv) extends AkkaSpecification with ElasticsearchTestKit {
```

Comment:
Can you add a test to check our retry logic?

I was thinking of something like this:
- Add a type mapping to a generic index
- Try inserting 2 docs, one that respects the mapping and one that doesn't.
- Ensure that only one doc is inserted.
- Remove the mapping
- Confirm the 2 docs were inserted correctly

Reply:
That is a good idea!

Done in a5ee658. This was done in a slightly different manner, using the "dynamic" property, because altering template fields is not allowed.